openupgrade_scripts/scripts/stock/14.0.1.1/post-migration.py

```diff
@@ -24,12 +24,12 @@ def merge_priorities(env):
 def recompute_stock_picking_scheduled_date(env):
     env.cr.execute(
         """
-        SELECT sp.id
+        SELECT DISTINCT sp.id
         FROM stock_picking sp
         JOIN stock_move sm ON sm.picking_id = sp.id
         WHERE sp.state NOT IN ('done', 'cancel')"""
     )
-    picking_ids = [pick[0] for pick in env.cr.fetchall()]
+    picking_ids = [pick for pick, *_ in env.cr.fetchall()]
```

Comment:
why?

Reply:
Original approach is "more efficent" because only unpack needed column.

In this case it doesn't matter because there is only one column, but if it's just a question of style it's better to get used to styles that offer the best performance. :wink:

Reply:
Well, I have to test it and you're completely right :)

```python
>>> timeit(lambda: [pick for pick, *_ in pickings], number=1)
0.7465282640041551
>>> timeit(lambda: [pick[0] for pick in pickings], number=1)
0.12576593900303124
```

I'd still prefer it when performance isn't an issue as a more expressive way of unpacking.

Anyway, I revert it ;)